crowbar_framework/app/models/service_object.rb

```diff
@@ -1300,105 +1300,41 @@ def apply_role(role, inst, in_queue)
       next if batch.empty?
       @logger.debug "batch #{batch.inspect}"
 
-      non_admin_nodes = []
-      admin_list = []
       batch.each do |node_name|
         # Run admin nodes a different way.
         if admin_nodes.include?(node_name)
           @logger.debug "#{node_name} is in admin_nodes #{admin_nodes.inspect}"
-          admin_list << node_name
           ran_admin = true
           next
         end
-        non_admin_nodes << node_name
       end
 
-      #
-      # XXX: We used to do this twice - do we really need twice???
-      # Yes! We do!  The system has some transient issues that are hidden
-      # but the double run for failing nodes.  For now, we will do this.
-      # Make this better one day.
-      #
+      @logger.debug(
+        "AR: Calling chef-client for #{role.name} on non-admin nodes #{batch.join(" ")}"
```

Comment:
So is this actually correct, now when you call `run_remote_chef_client` for all nodes, not just non-admin ones?